### PR TITLE
mBART support for run_summarization.py

### PR DESCRIPTION
**Pull request by banda-larga**
Added support for multilingual tokenizer and mBART for run_summarization.py

-> run_summarization.py not working with mBART

### CODE REVIEW

File: examples/pytorch/summarization/run_summarization.py [hunk 1]
```diff
@@ -114,6 +120,9 @@ class DataTrainingArguments:
     Arguments pertaining to what data we are going to input our model for training and eval.
     """
 
+    source_lang: str = field(default=None, metadata={"help": "Source language id for summarization."})
+    target_lang: str = field(default=None, metadata={"help": "Target language id for summarization."})
```
Comment (sgugger) [inline examples/pytorch/summarization/run_summarization.py:124]: If the language is different for the inputs and the outputs it is a translation task, not a summarization task, so there should only be one language argument.

File: examples/pytorch/summarization/run_summarization.py [hunk 2]
```diff
@@ -217,12 +226,24 @@ class DataTrainingArguments:
...
-        default=None, metadata={"help": "A prefix to add before every source text (useful for T5 models)."}
+        default="", metadata={"help": "A prefix to add before every source text (useful for T5 models)."}
+    )
+
+    forced_bos_token: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "The token to force as the first generated token after the :obj:`decoder_start_token_id`."
+            "Useful for multilingual models like :doc:`mBART <../model_doc/mbart>` where the first generated token "
+            "needs to be the target language token.(Usually it is the target language token)"
```
Comment (sgugger) [inline examples/pytorch/summarization/run_summarization.py:237]: Use MarkDown here, no need for obj or docs. Also the link should be resolved.

File: examples/pytorch/summarization/run_summarization.py [hunk 3]
```diff
@@ -436,14 +483,8 @@ def main():
         )
 
     def preprocess_function(examples):
-
-        # remove pairs where at least one record is None
-        inputs, targets = [], []
-        for i in range(len(examples[text_column])):
-            if examples[text_column][i] is not None and examples[summary_column][i] is not None:
-                inputs.append(examples[text_column][i])
-                targets.append(examples[summary_column][i])
```
Comment (sgugger) [inline examples/pytorch/summarization/run_summarization.py:445]: Why is this removed?

**Review by sgugger — COMMENTED** [review]
I don't think we should duplicate everything from the translation script, only take what is necessary for the summarization script to work with MBart.

File: examples/pytorch/summarization/run_summarization.py (hunk 3 above)
Comment (banda-larga) [reply examples/pytorch/summarization/run_summarization.py]: Yes my code was slightly old, fixed it

File: examples/pytorch/summarization/run_summarization.py [hunk 4]
```diff
@@ -406,6 +432,25 @@ def main():
...
+        # For multilingual translation models like mBART-50 and M2M100 we need to force the target language token
+        # as the first generated token. We ask the user to explicitly provide this as --forced_bos_token argument.
+        forced_bos_token_id = (
+            tokenizer.lang_code_to_id[data_args.forced_bos_token] if data_args.forced_bos_token is not None else None
+        )
+        model.config.forced_bos_token_id = forced_bos_token_id
+
+    # Get the language codes for input/target.
+    source_lang = data_args.lang.split("_")[0]
+    target_lang = data_args.lang.split("_")[0]
```
Comment (sgugger) [inline examples/pytorch/summarization/run_summarization.py:452]: Those are not used in the rest of the example if I'm not mistaken.

**Review by sgugger — APPROVED** [review]
Thanks for adapting! Can you address the last two comments and run `make style` on your branch to get rid of the quality check?

File: examples/pytorch/summarization/run_summarization.py (hunk 4 above)
Comment (banda-larga) [reply examples/pytorch/summarization/run_summarization.py]: Fixed!

**Comment by stas00** [comment]
So, it appears that this PR introduced an issue by forcing all uses to include `--lang`, @banda-larga would you like to fix it in a new PR and assert on lack of `--lang` only with mbart model type?

and while at it to integrate a better error message as proposed here https://github.com/huggingface/transformers/pull/15150

will then also need to revert this https://github.com/huggingface/transformers/pull/15149

and note to self: make sure 
```
RUN_SLOW=1 pytest tests/deepspeed/test_model_zoo.py::TestDeepSpeedModelZoo::test_zero_to_fp32_zero2_sum_pegasus
```
doesn't fail as this PR broke it.

Thanks.

**Comment by sgugger** [comment]
This is a tiny bit urgent, so not waiting to remove the two lines that break every existing command using `run_summarization`. I fixed that in [this commit](https://github.com/huggingface/transformers/commit/96881729ce83cfc8e5fa04c903ee4296ad17cfbb).

**Comment by stas00** [comment]
ok, so nothing else needs to be done. @banda-larga please ignore my comments above.
